### ISSUE TITLE
Smelting task does not work

### ISSUE THREAD

**Issue by doitliketyler** [LMBishop/Quests#443]
### Description of Bug

I have quests for every single "smeltable" item listed on the wiki (https://minecraft.fandom.com/wiki/Smelting?so=search) and not a single one works. Not ore, sand, iron items, ancient debris...nothing.

### Steps to Reproduce

Setup smelting quest, test

### Environment

- Minecraft version: 1.19.1
- Server software: Paper
- Quests version: 3.13.1

### Agreements

- [X] I am running the latest version of Quests
- [X] I am certain this issue is unique and a similar issue is not currently open
- [X] I am using a version of Minecraft which is supported by Quests

**Comment by LMBishop** [comment]
Can you send an example task and also enable the debugger and send what it outputs>

**Comment by doitliketyler** [comment]
Sure. The example task is below and the debugger said this when I smelted a Golden Boots:

[doitliketyler - smelting1goldenboots/task - type 'smelting']
Player smelted item
[doitliketyler - smelting1goldenboots/task - type 'smelting']
Item does not match required item, continuing...

tasks:
  smelt:
    amount: 1
    item: "GOLDEN_BOOTS"
    type: "smelting"
display:
  type: "GOLDEN_BOOTS"
  name: "Smelt Golden Boots"
  lore-normal:
    - "Task"
    - "Smelt 1 Golden Boots"
    - ""
    - "Progress"
    - "{smelt:progress}/1 Golden Boots"
placeholders:
  progress: "{smelt:progress}/1 Golden Boots"
  task: "Smelt 1 Golden Boots"
rewards:
  - "money give {player} 50"
options:
  category: "smelting"
  sort-order: 4

**Comment by Krakenied** [comment]
There is no way to check for the input item. You can only check for the output. In this case, the output will be a gold nugget.

**Comment by doitliketyler** [comment]
That is not true. Even just using Denizen you could write a script that flags when a player puts an item into a furnace, what the item is and what location is. Then by watching the smelting event you can get the source item, resulting item and furnace location and then cross-reference the previous flags.

**Comment by Krakenied** [comment]
Yes, definitely there is a way to fix it, but it won't remember the source item after server restarts. Also, It'll break when using with automatic (for example output only) hopper systems. It won't work with `hopper.disable-move-event: true` too. And even if we fixed it (tbh broke, not fixed), it would have major performance impact.
